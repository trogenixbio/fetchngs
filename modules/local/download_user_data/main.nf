process DOWNLOAD_USER_DATA {
    tag "$meta.id"
    label 'process_medium'

    conda "conda-forge::python=3.9.1"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/python:3.9--1' :
        'biocontainers/python:3.9--1' }"

    input:
    tuple val(meta), path(fastq, stageAs: 'fastq/*')

    output:
    tuple val(meta), path("*fastq.gz"), emit: fastq
    tuple val(meta), path("*md5")     , emit: md5
    path "versions.yml"               , emit: versions

    script:
    def args = task.ext.args ?: ''

    def echo_md5_single = meta.md5_1 ? "echo '${meta.md5_1}  ${meta.id}.fastq.gz'": "echo 'No md5sum available for ${meta.id}.fastq.gz'"

    def echo_md5_1 = meta.md5_1 ? "echo '${meta.md5_1}  ${meta.id}_1.fastq.gz'": "echo 'No md5sum available for ${meta.id}_1.fastq.gz'"
    def echo_md5_2 = meta.md5_2 ? "echo '${meta.md5_2}  ${meta.id}_2.fastq.gz'": "echo 'No md5sum available for ${meta.id}_2.fastq.gz'"

    def md5_1 = meta.md5_1 ? "md5sum -c": "touch "
    def md5_2 = meta.md5_2 ? "md5sum -c": "touch "

    if (meta.single_end) {
        """
        $echo_md5_single > ${meta.id}.fastq.gz.md5
        cp $fastq ${meta.id}.fastq.gz
        $md5_1 ${meta.id}.fastq.gz.md5

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            python: \$(python --version | sed 's/Python //g')
        END_VERSIONS
        """
    } else {
        """
        $echo_md5_1 > ${meta.id}_1.fastq.gz.md5
        cp ${fastq[0]} ${meta.id}_1.fastq.gz
        $md5_1 ${meta.id}_1.fastq.gz.md5

        $echo_md5_2 > ${meta.id}_2.fastq.gz.md5
        cp ${fastq[1]} ${meta.id}_2.fastq.gz
        $md5_2 ${meta.id}_2.fastq.gz.md5

        cat <<-END_VERSIONS > versions.yml
        "${task.process}":
            python: \$(python --version | sed 's/Python //g')
        END_VERSIONS
        """
    }
}
