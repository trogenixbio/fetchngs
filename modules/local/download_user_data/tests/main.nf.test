nextflow_process {

    name "Test process: DOWNLOAD_USER_DATA"
    script "../main.nf"
    process "DOWNLOAD_USER_DATA"

    test("Should run without failures - with s3 md5check") {

        when {
            process {
                """
                input[0] = [
                    [ id:'SRX9626017_SRR13191702', sample:'SRX9626017', single_end:false, md5_1: 'cfb2a2ee32c0bfb465f1287d441bea69', md5_2: 'ef99d9e789b8bba25e9788c4d4267b9c' ], // meta map
                    [
                        file('s3://ngi-igenomes/test-data/mag/ERR3201918_1.fastq.gz'),
                        file('s3://ngi-igenomes/test-data/mag/ERR3201918_2.fastq.gz')
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Should run without failures - with s3 no md5check") {

        when {
            process {
                """
                input[0] = [
                    [ id:'SRX9626017_SRR13191702', sample:'SRX9626017', single_end:false, md5_1: '', md5_2: '' ], // meta map
                    [
                        file('s3://ngi-igenomes/test-data/mag/ERR3201918_1.fastq.gz'),
                        file('s3://ngi-igenomes/test-data/mag/ERR3201918_2.fastq.gz')
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Should run without failures - with https no md5check") {

        when {
            process {
                """
                input[0] = [
                    [ id:'SRX9626017_SRR13191702', sample:'SRX9626017', single_end:false, md5_1: '', md5_2: '' ], // meta map
                    [
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/testdata/GSE110004/SRR6357070_1.fastq.gz'),
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/testdata/GSE110004/SRR6357070_2.fastq.gz')
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("Should run without failures - with https and md5check") {

        when {
            process {
                """
                input[0] = [
                    [ id:'SRX9626017_SRR13191702', sample:'SRX9626017', single_end:false, md5_1: 'd19f3c7a8b4016fa641458257b99a1a1', md5_2: 'd126cc66ed169f69af2c07da0dd73e02' ], // meta map
                    [
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/testdata/GSE110004/SRR6357070_1.fastq.gz'),
                        file('https://raw.githubusercontent.com/nf-core/test-datasets/rnaseq/testdata/GSE110004/SRR6357070_2.fastq.gz')
                    ]
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
