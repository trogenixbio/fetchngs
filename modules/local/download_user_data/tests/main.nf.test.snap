{
    "Should run without failures - with https and md5check": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "d19f3c7a8b4016fa641458257b99a1a1",
                            "md5_2": "d126cc66ed169f69af2c07da0dd73e02"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,e5f87bae86ec751571dbf3b1aa0c748a",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,dcdee33320a67180256dfd4bc4353173"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "d19f3c7a8b4016fa641458257b99a1a1",
                            "md5_2": "d126cc66ed169f69af2c07da0dd73e02"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,f6f450e9cb404c772457f9fa7345f1d9",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,e78a2b1172e8778d5833ab4e3653fdb4"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "d19f3c7a8b4016fa641458257b99a1a1",
                            "md5_2": "d126cc66ed169f69af2c07da0dd73e02"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,e5f87bae86ec751571dbf3b1aa0c748a",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,dcdee33320a67180256dfd4bc4353173"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "d19f3c7a8b4016fa641458257b99a1a1",
                            "md5_2": "d126cc66ed169f69af2c07da0dd73e02"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,f6f450e9cb404c772457f9fa7345f1d9",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,e78a2b1172e8778d5833ab4e3653fdb4"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T11:40:59.137333569"
    },
    "Should run without failures - with s3 md5check": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,ca9b49b42211256307d15b226a6082c6",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,9244c3a83ae8dae057b137f06c101030"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,1ab088890638dccd1d76d8483d330468",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,a1df8810780058996cc86a2fe9b7a27d"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,ca9b49b42211256307d15b226a6082c6",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,9244c3a83ae8dae057b137f06c101030"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,1ab088890638dccd1d76d8483d330468",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,a1df8810780058996cc86a2fe9b7a27d"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T11:20:32.7751797"
    },
    "Should run without failures - with s3 no md5check": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,ca9b49b42211256307d15b226a6082c6",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,9244c3a83ae8dae057b137f06c101030"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,fb789e5065dc1077479cc38dd5151017",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,ccf542a78280906897bee94f231ddb69"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,ca9b49b42211256307d15b226a6082c6",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,9244c3a83ae8dae057b137f06c101030"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,fb789e5065dc1077479cc38dd5151017",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,ccf542a78280906897bee94f231ddb69"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T11:25:28.893431867"
    },
    "Should run without failures": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,ca9b49b42211256307d15b226a6082c6",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,9244c3a83ae8dae057b137f06c101030"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,1ab088890638dccd1d76d8483d330468",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,a1df8810780058996cc86a2fe9b7a27d"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,ca9b49b42211256307d15b226a6082c6",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,9244c3a83ae8dae057b137f06c101030"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "cfb2a2ee32c0bfb465f1287d441bea69",
                            "md5_2": "ef99d9e789b8bba25e9788c4d4267b9c"
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,1ab088890638dccd1d76d8483d330468",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,a1df8810780058996cc86a2fe9b7a27d"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T11:11:52.810121054"
    },
    "Should run without failures - with https no md5check": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,e5f87bae86ec751571dbf3b1aa0c748a",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,dcdee33320a67180256dfd4bc4353173"
                        ]
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,fb789e5065dc1077479cc38dd5151017",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,ccf542a78280906897bee94f231ddb69"
                        ]
                    ]
                ],
                "2": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ],
                "fastq": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz:md5,e5f87bae86ec751571dbf3b1aa0c748a",
                            "SRX9626017_SRR13191702_2.fastq.gz:md5,dcdee33320a67180256dfd4bc4353173"
                        ]
                    ]
                ],
                "md5": [
                    [
                        {
                            "id": "SRX9626017_SRR13191702",
                            "sample": "SRX9626017",
                            "single_end": false,
                            "md5_1": "",
                            "md5_2": ""
                        },
                        [
                            "SRX9626017_SRR13191702_1.fastq.gz.md5:md5,fb789e5065dc1077479cc38dd5151017",
                            "SRX9626017_SRR13191702_2.fastq.gz.md5:md5,ccf542a78280906897bee94f231ddb69"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,08c1445fdb973c394f6d470bca322c27"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-04T11:33:21.500797463"
    }
}